### PR TITLE
Add version info to druid package and use it for CLI version output

### LINKED ISSUE #29: Add command-line option to show version

### PR DESCRIPTION
**Pull request by simonvanderveldt**
This uses https://pypi.org/project/setuptools-scm/ + https://pypi.org/project/setuptools-scm-git-archive/ which is a bit "magic", but from my testing this is the only way to get the version automatically with the following criteria:
- The only action necessary to update the version is `git tag`
- Release archives from GitHub should work and show the correct version
- During development it should be clear that a "development version" is being used. Ideally the shorthash + something like `dev` should be shown

Some output examples:
```
$ druid --version          
druid, version 0.1.2.dev17+g971de94
$ git tag -a 0.1.3 -m "thisisatest"
$ druid --version
druid, version 0.1.3
$ echo "foo" > test.txt
$ git add test.txt
$ druid --version
druid, version 0.1.4.dev0+g07eb4c8.d20191021
```

And when making a release archive (which is what GitHub does):
```
$ git tag -a 0.1.3 -m "thisisatest" 
$ git archive add-version-info | gzip > test.tgz
$ tar -axf test.tgz .git_archival.txt -O       
ref-names: HEAD -> add-version-info, tag: 0.1.3, origin/add-version-info
```
And when installing from this archive
```
$ pip install test.tgz
...
$ druid --version
druid, version 0.1.3
```

@csboling what do you think?

Fixes #29

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -21,11 +20,18 @@
...
+    setup_requires=[
+        "setuptools_scm",
+        "setuptools_scm_git_archive",
+    ],
     install_requires=[
         "Click>=7.0",
         "prompt-toolkit>=2.0.10",
         "pyserial>=3.4",
+        "setuptools_scm",
+        "setuptools_scm_git_archive",
```
Comment (simonvanderveldt) [inline setup.py]: We need to have these installed at runtime as well, otherwise the detection isn't possible.

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+.git_archival.txt  export-subst
```
Comment (simonvanderveldt) [inline .gitattributes]: This is some git "magic" that processed the filename on the left using the attribute on the right.
`export-subst` basically substitutes some placeholders, see https://git-scm.com/docs/gitattributes#_creating_an_archive for more details.

The result is something like
```
ref-names: HEAD -> add-version-info, tag: 0.1.3, origin/add-version-info
```

**Comment by trentgill** [comment]
This looks good to me, and I don't mind the extra packages now that we're using an automated setup with pip. I can't really comment on the implementation, but it's at least terse and reads cleanly to the uninitiated (read: myself).

Perhaps you could add a section to the readme which describes the necessary steps to make a new release? I know it's going to be automated at some point, but would be good to document the current situation.

**Comment by simonvanderveldt** [comment]
> Perhaps you could add a section to the readme which describes the necessary steps to make a new release? I know it's going to be automated at some point, but would be good to document the current situation.

Good point! I'll add that tomorrow.

**Comment by trentgill** [comment]
@simonvanderveldt lmk what I can do to test this out and then let's check it in!

**Comment by simonvanderveldt** [comment]
@trentgill I don't expect it to behave differently on different OSes but if you could give it a spin on Mac OS and Windows that would be nice (although @csboling might do that as well)

**Comment by trentgill** [comment]
Tested all my systems and it seems to be working well! Merging now.

**Comment by csboling** [comment]
I just learned like seven new things about pip and git from trying to understand how this works, awesome. Very tidy solution!

**Pull request merged**